manifests/elasticsearch_exporter.pp

```diff
@@ -0,0 +1,140 @@
+#
+#  [*init_style*]
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+#
+#  [*install_method*]
+#  Installation method: url or package (only url is supported currently)
+#
+#  [*manage_group*]
+#  Whether to create a group for or rely on external code for that
+#
+#  [*manage_service*]
+#  Should puppet manage the service? (default true)
+#
+#  [*manage_user*]
+#  Whether to create user or rely on external code for that
+#
+#  [*os*]
+#  Operating system (linux is the only one supported)
+#
+#  [*package_ensure*]
+#  If package, then use this for package ensure default 'latest'
+#
+#  [*package_name*]
+#  The binary package name - not available yet
+#
+#  [*purge_config_dir*]
+#  Purge config files no longer generated by Puppet
+#
+#  [*restart_on_change*]
+#  Should puppet restart the service on configuration change? (default true)
+#
+#  [*service_enable*]
+#  Whether to enable the service from puppet (default true)
+#
+#  [*service_ensure*]
+#  State ensured for the service (default 'running')
+#
+#  [*uri*]
+#  HTTP API address of an Elasticsearch node. (default "http://localhost:9200")
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+class prometheus::elasticsearch_exporter (
+  $arch                 = $::prometheus::params::arch,
+  $bin_dir              = $::prometheus::params::bin_dir,
+  $download_extension   = $::prometheus::params::elasticsearch_exporter_download_extension,
+  $download_url         = undef,
+  $download_url_base    = $::prometheus::params::elasticsearch_exporter_download_url_base,
+  $extra_groups         = $::prometheus::params::elasticsearch_exporter_extra_groups,
+  $extra_options        = '',
+  $group                = $::prometheus::params::elasticsearch_exporter_group,
+  $init_style           = $::prometheus::params::init_style,
+  $install_method       = $::prometheus::params::install_method,
+  $listen_address       = $::prometheus::params::elasticsearch_exporter_listen_address,
+  $manage_group         = true,
+  $manage_service       = true,
+  $manage_user          = true,
+  $os                   = $::prometheus::params::os,
+  $package_ensure       = $::prometheus::params::elasticsearch_exporter_package_ensure,
+  $package_name         = $::prometheus::params::elasticsearch_exporter_package_name,
+  $purge_config_dir     = true,
+  $restart_on_change    = true,
+  $service_enable       = true,
+  $service_ensure       = 'running',
+  $service_name         = 'elasticsearch_exporter',
+  $uri                  = $::prometheus::params::elasticsearch_exporter_uri,
+  $user                 = $::prometheus::params::elasticsearch_exporter_user,
+  $version              = $::prometheus::params::elasticsearch_exporter_version,
+) inherits prometheus::params {
+  $real_download_url    = pick($download_url,"${download_url_base}/download/v${version}/${package_name}-${version}.${os}-${arch}.${download_extension}")
+  validate_bool($purge_config_dir)
```

Comment:
Hi @rbestbmj, thanks for this PR! Can you replace those legacy functions with native puppet datatypes?

Reply:
@bastelfreak I've added the datatypes.